### DISCUSSION TITLE
What to do about "the cache artifact for <hash> was too large to upload within the timeout"

### DISCUSSION THREAD

**Discussion by NullVoxPopuli** [ducktors/turborepo-remote-cache#407] [Q&A]
what controls this message?
is there an environment variable to set?

**Comment by toretua** [comment]
I have same issue...
For some of the blobs i get this message:
 WARNING  the cache artifact for d6a63eff4c6deb9e was too large to upload within the timeout
 WARNING  the cache artifact for fb6dc9f4fa1e53b2 was too large to upload within the timeout
 WARNING  the cache artifact for 12a78c114ce555d8 was too large to upload within the timeout

AWS setup with Lambda behind ELB with cert and S3 as storage
The setup works for most of the blob, but some peek some limit and are not stored....

**Comment by EPMatt** [comment]
It looks like the timeout warning comes from the `vercel/turborepo` client. The timeout for the APIClient that uploads artifacts is set to 200 seconds:

https://github.com/vercel/turborepo/blob/7fb5b21fe3034c29affef8e35a8c90ee75c0cb67/crates/turborepo-cache/src/http.rs#L333-L339
```rust
        let api_client = APIClient::new(
            format!("http://localhost:{}", port),
            Some(Duration::from_secs(200)),
            None,
            "2.0.0",
            true,
        )?;
```

https://github.com/vercel/turborepo/blob/7fb5b21fe3034c29affef8e35a8c90ee75c0cb67/crates/turborepo-api-client/src/lib.rs#L576-L593
```rust
impl APIClient {
    /// Create a new APIClient.
    ///
    /// # Arguments
    /// `base_url` - The base URL for the API.
    /// `timeout` - The timeout for requests.
    /// `upload_timeout` - If specified, uploading files will use `timeout` for
    ///                    the connection, and `upload_timeout` for the total.
    ///                    Otherwise, `timeout` will be used for the total.
    /// `version` - The version of the client.
    /// `use_preflight` - If true, use the preflight API for all requests.
    pub fn new(
        base_url: impl AsRef<str>,
        timeout: Option<Duration>,
        upload_timeout: Option<Duration>,
        version: &str,
        use_preflight: bool,
    ) -> Result<Self> {
```

**Reply by EPMatt** [reply]
Upvote the feature request for configuring a custom upload timeout in the turbo client here:
https://github.com/vercel/turborepo/discussions/9472

**Comment by giancarlosisasi** [comment]
Hi guys,
So for now what could be the solution? just avoid using the cache? 😢

**Reply by NullVoxPopuli** [reply]
did you try bumping the upload timeout?
